src/Umbraco.Web/Unversion/UnversionConfig.cs

```diff
@@ -0,0 +1,97 @@
+
+        private ILogger _logger;
+
+        public UnversionConfig(ILogger logger)
+        {
+            _logger = logger;
+
+            ConfigEntries = new Dictionary<string, List<UnversionConfigEntry>>();
+
+            try
+            {
+                var appPath = HttpRuntime.AppDomainAppPath;
+                var configFilePath = Path.Combine(appPath, @"config\unversion.config");
+                LoadXmlConfig(string.Concat(configFilePath));
+            }
+            catch (Exception e)
+            {
+                _logger.Error<UnversionConfig>(e, "Error when parsing unversion.config.");
+            }
+
+        }
+
+        private void LoadXmlConfig(string configPath)
+        {
+            if (!File.Exists(configPath))
+            {
+                _logger.Warn<UnversionConfig>("Couldn't find config file {configPath}", configPath);
+                return;
+            }
+
+            var xmlConfig = new XmlDocument();
+            xmlConfig.Load(configPath);
+
+            foreach (XmlNode xmlConfigEntry in xmlConfig.SelectNodes("/unversionConfig/add"))
+                if (xmlConfigEntry.NodeType == XmlNodeType.Element)
+                {
+                    var configEntry = new UnversionConfigEntry
+                    {
+                        // If no doctype alias set on XML entry then its assume it's for all doctypes
+                        DocTypeAlias = xmlConfigEntry.Attributes["docTypeAlias"] != null
+                            ? xmlConfigEntry.Attributes["docTypeAlias"].Value
+                            : AllDocumentTypesKey
+                    };
+
+                    if (xmlConfigEntry.Attributes["rootXpath"] != null)
+                        configEntry.RootXPath = xmlConfigEntry.Attributes["rootXpath"].Value;
+
+                    if (xmlConfigEntry.Attributes["maxDays"] != null)
+                        configEntry.MaxDays = Convert.ToInt32(xmlConfigEntry.Attributes["maxDays"].Value);
+
+                    if (xmlConfigEntry.Attributes["maxCount"] != null)
+                        configEntry.MaxCount = Convert.ToInt32(xmlConfigEntry.Attributes["maxCount"].Value);
+
+                    if (xmlConfigEntry.Attributes["minCount"] != null)
+                        configEntry.MinCount = Convert.ToInt32(xmlConfigEntry.Attributes["minCount"].Value);
+
+                    if (!ConfigEntries.ContainsKey(configEntry.DocTypeAlias))
+                        ConfigEntries.Add(configEntry.DocTypeAlias, new List<UnversionConfigEntry>());
+
+                    ConfigEntries[configEntry.DocTypeAlias].Add(configEntry);
+                }
+        }
+    }
+
+    public class UnversionConfigEntry
+    {
+        public UnversionConfigEntry()
+        {
+            MaxDays = MaxCount = int.MaxValue;
+            MinCount = int.MinValue;
+        }
+
+        public string DocTypeAlias { get; set; }
+        public string RootXPath { get; set; }
```

Comment:
Do we really want this? How important is this? It adds quite a lot of complexity and seems overkill IMO. Versioning should just be based on document type and hopefully the configuration of each type would be part of the UI not part of config - since that would require adjusting the config (code) for each doc type created/changed instead of managing it all from the back office and being able to define defaults. The amount of work involved to make this config seems to outweigh the work for having this in the UI. But specifying this XPath means we could never have it in the UI.